maintenance-branch

```diff
@@ -1 +1 @@
-4-0-dev
\ No newline at end of file
+drop-old-rubies
```

Comment:
⚠️ This change is temporary and needs to be removed before merging, or soon after ⚠️

Reply:
Why is this required?

Reply:
I want to make sure that all those changes to our four repos work together.
4-0-dev branches have almost no differences (apart from Gemfile and CI files) from 3.x.
Because of
```
# pin to major/minor ignoring patch
```
I assume that compatibility of 4.0 with 4.0.x of our other gems is more important than with 3.x (that 4-0-dev didn't diverge much from at this moment).

It would be a nice bonus if `rspec-mocks` worked with `rspec-support` 3.9, and it probably does, but I was not too concerned about this compatibility.

Reply:
Removing old ruby support should be entirely compatible with 3.9, we're not adding anything so working with the `main` branch is confidence boosting and is a benefit.

Reply:
It was 4-0-dev actually in the past. Reverted.
Hoping for a green build.